lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -3707,3 +3733,75 @@ def get_test_data(delta=0.05):
     Y = Y * 10
     Z = Z * 500
     return X, Y, Z
+
+
+class _Quaternion:
+    """
+    Quaternions
+    consisting of scalar, along 1, and vector, with components along i, j, k
+    """
+
+    def __init__(self, scalar, vector):
+        self.scalar = scalar
+        self.vector = np.array(vector)
+
+    def __neg__(self):
+        return self.__class__(-self.scalar, -self.vector)
+
+    def __mul__(self, other):
+        """
+        Product of two quaternions
+        i*i = j*j = k*k = i*j*k = -1
+        """
+        return self.__class__(
+            self.scalar*other.scalar - np.dot(self.vector, other.vector),
+            self.scalar*other.vector + self.vector*other.scalar
+            + np.cross(self.vector, other.vector))
+
+    def __eq__(self, other):
+        return (self.scalar == other.scalar) and (self.vector == other.vector).all
+
+    def __repr__(self):
+        return "_Quaternion({}, {})".format(repr(self.scalar), repr(self.vector))
+
+    @classmethod
+    def from_to(cls, r1, r2):
+        """The quaternion that rotates vector r1 to vector r2."""
+        k = np.cross(r1, r2)
+        nk = np.linalg.norm(k)
+        th = np.arctan2(nk, np.dot(r1, r2))
+        th = th/2
+        if nk == 0:  # r1 and r2 are parallel or anti-parallel
+            if np.dot(r1, r2) < 0:
+                q = cls(0, [1, 0, 0])  # axis of the circle that r1 and r2 are on
+            else:
+                q = cls(1, [0, 0, 0])  # = 1, no rotation
+        else:
+            q = cls(math.cos(th), k*math.sin(th)/nk)
+        return q
+
+    @classmethod
+    def from_cardan_angles(cls, elev, azim, roll):
```

Comment:
As another commenter pointed out, cardan angles are a bit too obscure to be a meaningful name for people reading the code, and also a bit too general to be worth naming them that way. I would refer to them as "view" angles, since we only ever have a single elev/azim/roll convention and that's how we refer to them elsewhere in the docs.

Reply:
You may consider Cardan angles obscure, but https://en.wikipedia.org/wiki/Cardan_angle takes you to an informative page. In contrast, the page "View angles" does not exist, wikipedia informs us. Just sayin'.

If you really can't stomach those Cardan angles, then I'm willing to consider `from_azimuth_elevation_roll()`, and call a spade a spade. It is longer, though. And I am unwilling to compromise on the order of the rotation angles.

Reply:
> Well, that does redirect you to wikipedia's "Euler Angles" page :) .

Of course it does. And it explains that there are Tait-Bryan angles too, that's useful to know. And if you look a little bit down the page, then you will see the Cardan angles mentioned. The wikipedia reference for Cardan angles points to that page for a reason.

Reply:
> and there they are unambiguous about _which_ euler/cardan angles rotation ordering is being used: https://matplotlib.org/stable/api/toolkits/mplot3d/view_angles.html

Well it is certainly a nice and helpful diagram; however, it does not give you any advance warning how strange this angle direction convention is, with the handedness of the azimuth rotation opposite to that of the elevation and roll. And while the angle direction may be a common convention, there does not appear to be such a consensus about the actual angle (https://www.mathworks.com/help/matlab/creating_plots/setting-the-viewpoint-with-azimuth-and-elevation.html) - an omission lying there in wait to catch the unwary.

Reply:
> Again, easy to change later since this is all a private interface.

Ah, then the cardan angles naming will serve just as well, for just a private interface. The elev-azim-roll cure would be worse than the disease.